### DISCUSSION TITLE
failed to accept an incoming connection

### DISCUSSION THREAD

**Discussion by alanmiller** [drakkan/sftpgo#1377] [Q&A]
I setup SFTPGo and verified the user's credentials work locally (same subnet) but when my customer tries to connect from his network we're seeing these 2 errors in the sftpgo.log file.

```
{
    "level": "debug",
    "time": "2023-08-02T19:25:34.786",
    "sender": "sftpd",
    "message": "failed to accept an incoming connection from ip \"1.2.3.4\": read tcp 10.20.30.40:22->1.12.3.4:6796: wsarecv: A connection attempt failed because the connected party did not properly respond after a period of time, or established connection failed because connected host has failed to respond."
}
{
    "level": "debug",
    "time": "2023-08-02T19:25:34.786",
    "sender": "connection_failed",
    "client_ip": "146.126.61.241",
    "username": "",
    "login_type": "no_auth_tried",
    "protocol": "SSH",
    "error": "read tcp 10.20.30.40:22->1.2.3.4:6796: wsarecv: A connection attempt failed because the connected party did not properly respond after a period of time, or established connection failed because connected host has failed to respond."
}
```

From the customer side, the connection just times out.
My server is protected via an ip whitelist but I see the customers ip in the error log so I don't know what's blocking the connection from getting established.

Any ideas?

**Comment by samuelperezh** [comment]
> I setup SFTPGo and verified the user's credentials work locally (same subnet) but when my customer tries to connect from his network we're seeing these 2 errors in the sftpgo.log file.
> 
> 
> 
> ```
> 
> {
> 
>     "level": "debug",
> 
>     "time": "2023-08-02T19:25:34.786",
> 
>     "sender": "sftpd",
> 
>     "message": "failed to accept an incoming connection from ip \"1.2.3.4\": read tcp 10.20.30.40:22->1.12.3.4:6796: wsarecv: A connection attempt failed because the connected party did not properly respond after a period of time, or established connection failed because connected host has failed to respond."
> 
> }
> 
> {
> 
>     "level": "debug",
> 
>     "time": "2023-08-02T19:25:34.786",
> 
>     "sender": "connection_failed",
> 
>     "client_ip": "146.126.61.241",
> 
>     "username": "",
> 
>     "login_type": "no_auth_tried",
> 
>     "protocol": "SSH",
> 
>     "error": "read tcp 10.20.30.40:22->1.2.3.4:6796: wsarecv: A connection attempt failed because the connected party did not properly respond after a period of time, or established connection failed because connected host has failed to respond."
> 
> }
> 
> ```
> 
> 
> 
> From the customer side, the connection just times out.
> 
> My server is protected via an ip whitelist but I see the customers ip in the error log so I don't know what's blocking the connection from getting established.
> 
> 
> 
> Any ideas?
> 
> 
> 
> 

Hi! I'm having the same issue right now, did you solve it?
